pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler.go

```
@@ -0,0 +1,473 @@
+		return "", "", err
+	}
+	componentConfigYAML = strings.Replace(componentConfigYAML, kubeschedulerv1alpha1.SchemeGroupVersion.String(), apiVersion, -1)
+
+	return componentConfigYAML, utils.ComputeChecksum(componentConfig), nil
+}
+
+func (k *kubeScheduler) computeCommandLineFlags(port int32) []string {
+	var command []string
+
+	if k.versionConstraints.k8sGreaterEqual117 {
+		command = append(command, "/usr/local/bin/kube-scheduler")
+	} else if k.versionConstraints.k8sGreaterEqual115 {
+		command = append(command, "/hyperkube", "kube-scheduler")
+	} else {
+		command = append(command, "/hyperkube", "scheduler")
+	}
+
+	command = append(command, fmt.Sprintf("--config=%s/%s", volumeMountPathConfig, DataKeyComponentConfig))
+
+	if k.versionConstraints.k8sGreaterEqual113 {
+		command = append(command,
+			fmt.Sprintf("--authentication-kubeconfig=%s/%s", volumeMountPathKubeconfig, secrets.DataKeyKubeconfig),
+			fmt.Sprintf("--authorization-kubeconfig=%s/%s", volumeMountPathKubeconfig, secrets.DataKeyKubeconfig),
+			fmt.Sprintf("--client-ca-file=%s/%s", volumeMountPathServer, secrets.DataKeyCertificateCA),
+			fmt.Sprintf("--tls-cert-file=%s/%s", volumeMountPathServer, secrets.ControlPlaneSecretDataKeyCertificatePEM(common.KubeSchedulerServerName)),
+			fmt.Sprintf("--tls-private-key-file=%s/%s", volumeMountPathServer, secrets.ControlPlaneSecretDataKeyPrivateKey(common.KubeSchedulerServerName)),
+			fmt.Sprintf("--secure-port=%d", port),
+			"--port=0",
+		)
+	}
+
+	if k.config != nil {
+		command = append(command, kutil.FeatureGatesToCommandLineParameter(k.config.FeatureGates))
+	}
+	if k.versionConstraints.k8sEqual110 {
+		command = append(command, kutil.FeatureGatesToCommandLineParameter(map[string]bool{"PodPriority": true}))
+	}
+
+	command = append(command, "--v=2")
+	return command
+}
+
+func (k *kubeScheduler) computeVersionConstraints() (*kubeSchedulerVersionConstraints, error) {
+	k8sEqual110, err := versionutils.CompareVersions(k.version, "~", "1.10")
```

I'm thinking about how this computation and the `kubeSchedulerVersionConstraints` construct can be simplified to reduce the amount of explicit error handling. Also this will parse the Shoot version for every constraint and each constraint for every shoot being reconciled...

WDYT about:
- "pre-computing" the semver constraints in an `init` func (panicking on error, as this is a programmer level error, maybe with `utilruntime.Must`)
- and parsing the shoots version to a `*semver.Version` when constructing the `Shoot` operation object?

Constraints can then be checked by just calling the `Check` func (which doesn't return an error).
This would slim down this component and the `New` func also wouldn't need to return an error. (Also we would have implicit test coverage for the constraints 😉 )